lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -59,6 +59,10 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
             // Get a reference to the actual signal to which the XMR will be
             // generated.
             Value xmrDef = send.getBase();
+            if (send.getType().getType().getBitWidthOrSentinel() == 0) {
+              markForRemoval(send);
+              return success();
+            }
```

Comment:
The check here is effectively repeated multiple times. Consider adding a lambda for `isZeroWidth`.

Reply:
Thanks Schuyler, added a function for this check.